community/logging/src/main/java/org/neo4j/logging/RotatingFileOutputStreamSupplier.java

```diff
@@ -227,10 +236,10 @@ public void run()
         {
             rotationExecutor.execute( runnable );
         }
-        catch ( RejectedExecutionException e )
+        catch ( Exception e )
         {
-            // Run directly instead
-            runnable.run();
+            rotationListener.rotationError( e, outRef.get() );
```

Comment:
I'm painting a bike shed here, but I see three issues:

1) We should use JobScheduler for this task
2) `RejectedExecutionException` should signify an illegal state (shutdown) or serious problem (too many tasks queued) and should not be treated as a failure to rotate, rather `ThisShouldNotHappenError` or something.
3) Only `ExecutionException` should be caught and translated into a `rotationError`